File: Algorithm/Recursion/Solution/N과 M(3) - boj/BOJ15651.py
def printSequence(bucket):
    if len(bucket) == M:
        print(' '.join(map(str, bucket)))
        return

    for i in range(N):
        bucket.append(i + 1)
        printSequence(bucket)
        bucket.pop()


N, M = map(int, input().split())
printSequence([])

File: Algorithm/Recursion/Solution/N과 M(3) - boj/BOJ15651.c
#include <stdio.h>

int N, M;

void printSequence(int cnt, int bucket[])
{
    if (cnt == M)
    {
        for (int i = 0; i < M; ++i)
        {
            printf("%d ", bucket[i]);
        }

        printf("\n");
        return;
    }

    for (int i = 1; i <= N; ++i)
    {
        bucket[cnt] = i;
        printSequence(cnt + 1, bucket);
    }
}

int main(void)
{
    scanf("%d %d", &N, &M);
    int bucket[7];
    printSequence(0, bucket);
    return 0;
}